sdk/identity/identity/src/util/tenantIdUtils.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ALL_TENANTS, DeveloperSignOnClientId } from "../constants";
+import { CredentialLogger, formatError } from "./logging";
+import { GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * @internal
+ */
+export function checkTenantId(logger: CredentialLogger, tenantId: string): void {
+  if (!tenantId.match(/^[0-9a-zA-Z-.:/]+$/)) {
+    const error = new Error(
+      "Invalid tenant id provided. You can locate your tenant id by following the instructions listed here: https://docs.microsoft.com/partner-center/find-ids-and-domain-names."
+    );
+    logger.info(formatError("", error));
+    throw error;
+  }
+}
+
+/**
+ * @internal
+ */
+export function resolveTenantId(
+  logger: CredentialLogger,
+  tenantId?: string,
+  clientId?: string
+): string {
+  if (tenantId) {
+    checkTenantId(logger, tenantId);
+    return tenantId;
+  }
+  if (!clientId) {
+    clientId = DeveloperSignOnClientId;
+  }
+  if (clientId !== DeveloperSignOnClientId) {
+    return "common";
+  }
+  return "organizations";
+}
+
+function createConfigurationErrorMessage(tenantId: string) {
+  return `The current credential is not configured to acquire tokens for tenant ${tenantId}. To enable acquiring tokens for this tenant add it to the AdditionallyAllowedTenants on the credential options, or add "*" to AdditionallyAllowedTenants to allow acquiring tokens for any tenant.`;
+}
+
+/**
+ * Of getToken contains a tenantId, this functions allows picking this tenantId as the appropriate for authentication,
+ * unless multitenant authentication has been disabled through the AZURE_IDENTITY_DISABLE_MULTITENANTAUTH (on Node.js),
+ * or unless the original tenant Id is `adfs`.
+ * @internal
+ */
+export function processMultiTenantRequest(
+  tenantId?: string,
+  getTokenOptions?: GetTokenOptions,
+  additionallyAllowedTenantIds: string[] = []
+): string | undefined {
+  let resolvedTenantId: string | undefined;
+  if (process.env.AZURE_IDENTITY_DISABLE_MULTITENANTAUTH) {
```

Comment:
if this the only node-specific thing? These two files have a lot of overlap to keep in sync so maybe pull out resolving this environment variable into something with a node/browser split and unify tenantIdUtils?

Reply:
Yes, only node.  Refactored into a local TS file and re-exported